### ISSUE TITLE
Package heap_profiler

### ISSUE THREAD

**Issue by benshayden** [catapult-project/catapult#4068]
The Histogram-based Python [heap_profiler](https://chromium-review.googlesource.com/c/catapult/+/754453) currently has the following signature:
`Profile(root, label=None, html_filename=None, html_stream=None, vulcanized_viewer=None, reset_results=False, diagnostics_callback=None): HistogramSet`
This large number of optional parameters supports several different use-cases that, for the purposes of this bug, break down into 2 categories: with a catapult working directory (including use-cases with access to `chromium/src` such as `tools/perf/run_benchmark`), and without a working directory (including the appengine use-case such as chromeperf dashboard). When a catapult working directory is not available, Profile() requires `vulcanized_viewer` in order to render the results.html. Users can create the vulcanized viewer using [`VulcanizeHistogramsViewer()`](https://github.com/catapult-project/catapult/blob/master/tracing/tracing_build/vulcanize_histograms_viewer.py). Also, the heap_profiler requires the [histogram_set](https://github.com/catapult-project/catapult/blob/master/tracing/tracing/value/histogram_set.py) module and all its dependencies.

Between the vulcanized viewer and the dependencies, it might be difficult for some potential users to assemble all of the requirements by hand. If the viewer and dependencies were packaged into a Python egg, that might be easier to use. If the viewer were packaged with the code, then users would not need to pass it to the Profile() function, which would allow simplifying its signature by removing the vulcanized_viewer parameter. Also, it would presumably be easier to upload the heap_profiler package to appengine, since the user would not need to VulcanizeHistogramsViewer().

@nedn Is that an accurate summary of our discussion?

@simonhatch @eakuefner

**Comment by benshayden** [comment]
Relatedly, if the chromeperf dashboard provides a histograms viewer service #4005 , that could be used to simplify the process of rendering the Histograms. For example, Profile() could upload the json to cloudstorage and render a link to the viewer service, or else maybe results.html could iframe the viewer service and postMessage the JSON into the iframe.

**Comment by nedn** [comment]
Yes, this is the accurate summary of our discussion. To be clear, we gotta balance between getting thing right & make progress.

I would advocate for the following compromise to land the code:
1. Support vulcanizing using catapult dependencies only (this is because we don't use `heap_profiler` for anything that can't do vulcanizing yet)
2. Split to two methods, 1 create viewer by doing vulcanization & 1 create viewer based on existing vulcanized file. This is so that when you move forward with supporting 1 only, you have less headache deprecate the non supported code & don't have to change the APIs.
